cmds/core/ls/ls.go

```
@@ -37,15 +37,15 @@ var (
 	quoted    = flag.BoolP("quote-name", "Q", false, "quoted")
 	recurse   = flag.BoolP("recursive", "R", false, "equivalent to findutil's find")
 	classify  = flag.BoolP("classify", "F", false, "append indicator (one of */=>@|) to entries")
+	help      = flag.BoolP("help", "H", false, "show usage")
 )
 
 func listName(stringer ls.Stringer, d string, w io.Writer, prefix bool) error {
 	return filepath.Walk(d, func(path string, osfi os.FileInfo, err error) error {
 		// Soft error. Useful when a permissions are insufficient to
 		// stat one of the files.
 		if err != nil {
-			log.Printf("%s: %v\n", path, err)
-			return nil
+			return fmt.Errorf("%v", err)
```

You can simply return the error is you don't add any information via Errorf to is. Doing a %v simply takes the raw contents of the error without modification and wraps it in a new error object with the same contents.